Source/background.js

```diff
@@ -76,31 +60,111 @@ function translateOneRequestPerFewWords(words, prefs, callback) {
     translateORPFWRec(concatWordsArray, 1, cWALength, tMap, prefs, callback);
 }
 
+var translator = {
+    translate: function(prefs, word) {}
+};
+
+var googleTranslator = function() {};
+googleTranslator.prototype = Object.create(translator);
+
+var yandexTranslator = function() {};
+yandexTranslator.prototype = Object.create(translator);
+
+/**
+ * @desc Constructs google translate url
+ * @params User saved preferences
+ * @params word to be translated
+ * @return Google Translate query from source to destination language
+ */
+googleTranslator.prototype.translate = function(prefs, word, callback){
+    var url = 'http://translate.google.com/translate_a/t?client=f&otf=1&pc=0&hl=en';
+    var sL = prefs["sourceLanguage"];
+    if(sL != 'auto') {
+        url += '&sl=' + sL;
+    }
+    url += '&tl=' + prefs["targetLanguage"];
+    url += '&text=';
+    url += word;
+
+    getData(url, function(result){
+        var tMap = {}
+        for (i in result.sentences) {
+            var orig = result.sentences[i].orig;
+            var origT = orig.substring(0,orig.length - 1);
+            var trans = result.sentences[i].trans;
+            var transT = trans.replace(/[.\u3002]/,""); // removes punctuation
+            tMap[origT] = transT;
+        }
+        callback(tMap);
+    });
+}
+
+yandexTranslator.prototype.translate = function(prefs, word, callback){
+    var apikey = "trnsl.1.1.20150429T202526Z.d63a1483e719555a.852f432bad29a4628c1d1694b6114c0efb22f183"
```

Comment:
There are two option one is to let user generate the key and add it in settings, other is to have a dedicated api key for the extension, since the key is a secret maybe it is better to update it before publishing and not have it published in github.
From what i can figure out the limit is 1,000,000 characters per day which should be fine and i guess we can go with a single dedicated key for extension for now. If you want to generate a key you can do it from here https://tech.yandex.com/keys/get/?service=trnsl?service=trnsl